web/source/osk/visualKeyboard.ts

```diff
@@ -113,11 +138,23 @@ namespace com.keyman.osk {
         this.fontFamily='';
       }
 
-      let divLayerContainer = this.deviceDependentLayout(keyboard, device.formFactor as utils.FormFactor);
+      // Now to build the actual layout.
+      const formFactor = device.formFactor as utils.FormFactor;
+      let layoutKeyboard = keyboard;
+      if(!layoutKeyboard) {
+        // May occasionally be null in embedded contexts; have seen this when iOS engine sets
+        // keyboard height during change of keyboards.
+        layoutKeyboard = new keyboards.Keyboard(null);
+      }
 
+      this.layerGroup = new OSKLayerGroup(this, layoutKeyboard, formFactor);
+
+      // Now that we've properly processed the keyboard's layout, mark it as calibrated.
+      keyboard.markLayoutCalibrated(formFactor);
```

Comment:
This is what was left of `deviceDependentLayout` after the rest of its construction behavior was encapsulated within the new layout-hierarchy classes.